File: frontend/src/app/core/query/indicator-point-report.lookup.ts
import { Bucket } from "../model/bucket/bucket.model";
import { Metric } from "../model/metic/metric.model";
import { IndicatorPointLookup } from "./indicator-point.lookup";
import {Lookup} from '../../../common/model/lookup';

export class IndicatorPointReportLookup  implements IndicatorPointReportLookupFilter{
    filters: IndicatorPointLookup;
    metrics: Metric[];
    bucket: Bucket;
    isRawData?: boolean;
    rawDataRequest?: RawDataRequest;

	constructor() {
	}
}

export interface IndicatorPointReportLookupFilter {
    filters: IndicatorPointLookup;
	metrics: Metric[];
    bucket: Bucket;
}


export interface RawDataRequest{
    keyField: string;
    valueField: string;
    page?: Lookup.Paging;
    order: Lookup.Ordering;
}

File: backend/sti-viewer/src/main/java/gr/cite/intelcomp/stiviewer/model/elasticreport/IndicatorPointReportLookup.java
package gr.cite.intelcomp.stiviewer.model.elasticreport;

import gr.cite.intelcomp.stiviewer.elastic.query.lookup.IndicatorPointLookup;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;
import java.util.List;
import java.util.UUID;

public class IndicatorPointReportLookup {
	@Valid
	@NotNull(message = "{validation.empty}")
	private IndicatorPointLookup filters;
	@Valid
	private List<Metric> metrics;
	@Valid
	private Bucket bucket;
	private Boolean isRawData;
	@Valid
	private RawDataRequest rawDataRequest;

	public List<Metric> getMetrics() {
		return metrics;
	}

	public void setMetrics(List<Metric> metrics) {
		this.metrics = metrics;
	}

	public Bucket getBucket() {
		return bucket;
	}

	public void setBucket(Bucket bucket) {
		this.bucket = bucket;
	}

	public IndicatorPointLookup getFilters() {
		return filters;
	}

	public void setFilters(IndicatorPointLookup filters) {
		this.filters = filters;
	}

	public Boolean getIsRawData() {
		return isRawData;
	}

	public void setIsRawData(Boolean rawData) {
		isRawData = rawData;
	}

	public RawDataRequest getRawDataRequest() {
		return rawDataRequest;
	}

	public void setRawDataRequest(RawDataRequest rawDataRequest) {
		this.rawDataRequest = rawDataRequest;
	}
}
